File: src/1/maraton.java
import java.util.*;

class Solution {
    public String solution(String[] participant, String[] completion) {
        String answer = "";
        HashMap<String,Integer> map = new HashMap<String,Integer>();

        for(int i = 0 ; i<participant.length; i++) {
            if(!map.containsKey(participant[i])) {
                map.put(participant[i], 1);
            }else {
                int c = map.get(participant[i]);
                map.put(participant[i],c+1);
            }
        }


        for(int i=0; i<completion.length; i++) {
            if(map.containsKey(completion[i])) {
                if(map.get(completion[i]) == 1) {
                    map.remove(completion[i]);
                }else {
                    int c = map.get(completion[i]);
                    map.put(completion[i],c-1);
                }
            }
        }

        int len = map.keySet().toString().length();
        answer = map.keySet().toString().substring(1,len-1);
        return answer;
    }
}

File: src/1/maraton.kt
package `1`
import kotlin.collections.HashMap

class MarathonKt{
    fun solution(participant:Array<String> , completion:Array<String>) :String{
        var answer =""
        var map : HashMap<String,Int> = HashMap<String,Int>()
        for(i in participant.indices){
            if(map.containsKey(participant[i])){
                map.set(participant[i], map[participant[i]]!! + 1)
            }else{
                map.set(participant[i],1)
            }

        }
        for(i in completion.indices){
            if(map.containsKey(completion[i])){
                map.set(completion[i], map[completion[i]]!! - 1)
                if(map.get(completion[i]) == 0){
                    map.remove(completion[i])
                }
            }
        }

        map.forEach { k, v -> answer+=k }
        return answer
    }
}

fun main (args:Array<String>){
    var participant:Array<String> = arrayOf("mislav", "stanko","mislav", "ana")
    var completion:Array<String> = arrayOf("stanko", "ana", "mislav")
    print(MarathonKt().solution(participant,completion))
}